misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -201,17 +207,40 @@ private void createReport(final IBundleCoverage bundleCoverage, ModuleCoverage m
 
     private List<Path> filterPaths(Collection<Path> pathCollection) {
         List<Path> filteredPathList = new ArrayList<>();
-
+        List<Path> depsList = getDependencyJarPathsList();
         for (Path path : pathCollection) {
-            if (path.toString().contains(this.module.project().sourceRoot().toString()) &&
-                    path.toString().contains(target.cachesPath().toString())) {
+            if (!depsList.contains(path)) {
                 filteredPathList.add(path);
             }
         }
-
         return filteredPathList;
     }
 
+    private List<Path> getDependencyJarPathsList() {
+        List<Path> depPaths = new ArrayList<>();
+        JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(module.packageInstance().getCompilation(),
```

Comment:
We already create the JBallerinabackEnd in RunTestsTask. You can pass it on to CoverageReport and reuse it.